my_enumerable.rb

```diff
@@ -0,0 +1,154 @@
+
+# rubocop:disable all
+module Enumerable
+  def my_each
+    for item in self
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    for item in self
+      yield(item, index(item))
+    end
+  end
+
+  def my_select
+    for element in self
+      yield(element)
+   end
+ end
+
+  def my_any?(args = nil)
+    result = true
+    if !block_given? && args.nil?
+      empty? ? result = false : nil
+      my_each do |item|
+        if is_a? Hash
+          result = false if item[1] == false || item[1].nil?
+        elsif item.nil?
+          result = false
+        else
+          result = false if item == false || item.nil?
+        end
+      end
+      result
+    else
+      my_select { |key, value| yield(key, value) } if is_a? Hash
+      my_each { |key| yield(key) }
+    end
+  end
+
+  def my_all?(args = nil)
+    result = true
+    unless args.nil?
+      if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+      else
+        my_each do |_value|
+        break result = false unless item.class == args end
+      end
+    end
+    unless block_given?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == false || item[1].nil? end
+      else
+        my_each do |item|
+        result = false if item == false || item.nil? end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = false if yield(item, value) == false || yield(item, value).nil? end
+      else
+        my_each do |value|
+          break result = false if yield(value) == false || yield(value).nil?
+        end
+      end
+    end
+    result
+  end
+
+  def my_none?(args = nil)
+    result = true
+    unless args.nil?
+      if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+      else
+        my_each do |_value|
+        break result = false unless item.class == args end
+      end
+    end
+    unless block_given?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == true end
+      else
+        my_each do |item|
+        result = false if item == true end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        p yield(item, value) == true
+        break result = false if yield(item, value) == true end
+      else
+        my_each do |value|
+          break result = false if yield(value) == true
+        end
+      end
+    end
+    result
+  end
```

Comment:
- [ ] `my_none?`:

- When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello'].my_none?(5)` - should return `true`
- when a class is passed as an argument returns true if none of the collection is a member of such class
`p [3,3,3,3].my_none?(String)`  should return `true`
- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
`p ['hello','milk','milo'].my_none?(/z/)` should return `true`